.github/workflows/lint-test-build.yml

```diff
@@ -0,0 +1,173 @@
+name: Lint, Test & Build
+
+on:
+  pull_request:
+    branches:
+      - main
+      - feature/*
+  push:
+    branches:
+      - main
+      - feature/*
+
+concurrency:
+  group: lint-test-build-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  lint-test-rust:
+    name: Run Lint + Test Rust
+    runs-on: ubuntu-latest
+    container: ghcr.io/scuffletv/build:1.67.1
+    services:
+      postgres:
+        image: postgres:15.2
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: postgres
+        ports:
+          - 5432:5432
+
+    env:
+      DATABASE_URL: postgres://postgres:postgres@postgres:5432/scuffle
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install Node
+        uses: actions/setup-node@v3
+        with:
+          cache: "yarn"
+          node-version: 18
+
+      - name: Set up cargo registry cache
+        uses: actions/cache@v3
+        continue-on-error: false
+        with:
+          path: |
+            /usr/local/cargo/registry/index/
+            /usr/local/cargo/registry/cache/
+            /usr/local/cargo/git/db/
+          key: musl-rust-registry-${{ hashFiles('**/Cargo.lock', '**/Cargo.toml') }}
+          restore-keys: |
+            musl-rust-registry-
+
+      - name: Set up cargo target cache
+        uses: actions/cache@v3
+        continue-on-error: false
+        with:
+          path: |
+            target/
+          key: musl-rust-target-lint-test-rust-${{ hashFiles('**/Cargo.lock', '**/Cargo.toml') }}
+          restore-keys: |
+            musl-rust-target-lint-test-rust-
+
+      - name: Install sqlx-cli
+        run: |
+          curl -L https://github.com/ScuffleTV/ci-binaries/releases/download/sqlx-cli/sqlx-cli.tar.gz | tar -xz -C $CARGO_HOME/bin
```

Comment:
Pretty sure this is not how versioning is suppose to work. If we want to have a single repo for CI binaries we should version the packages. Maybe like usual (as Cargo does `x.y.z`) with the package name as `prerelease` according to SemVer. So sqlx-cli v0.6.2 would have the tag `0.6.2-sqlx-cli` (the `v` prefix is optional idk). Not sure, this still sucks.
My point is, the same commit should CI with the same result. So CI shouldn't rely on stuff that changes out-of-sync with the project git history. For example a branch instead of a commit, or a "latest" release for something (a new release will come out in the future and old builds can become unreproducible).

Reply:
I honestly, want the latest one, unless we change it in the other repo.

Reply:
We can version lock that repo if u want this is a fix because of how shit this sqlx-cli is where there are no precompiled binaries anywhere.